### ISSUE TITLE
Support for custom queries over JDBC

### ISSUE THREAD

**Issue by laneb** [TresAmigosSD/SMV#723]
Once support reading from and writing to JDBC is merged into master, we should enable users to provide custom queries as we do with `SmvHiveTable`. It appears that this can be accomplished by treating the user's query as a subquery and substituting it for the table name when reading.

See "Pushing down the whole query into MySQL" in https://www.percona.com/blog/2016/08/17/apache-spark-makes-slow-mysql-queries-10x-faster/ for an example.

**Comment by laneb** [comment]
Because this approached isn't documented or officially supported by Spark, this will not be available but not officially supported by SMV.
